core/textile/client.go

```diff
@@ -402,7 +402,9 @@ func (tc *textileClient) initialize(ctx context.Context) error {
 		return err
 	}
 
-	tc.sync.NotifyBucketStartup(defaultPersonalBucketSlug)
+	if tc.sync != nil {
+		tc.sync.NotifyBucketStartup(defaultPersonalBucketSlug)
```

Comment:
You know, I'm thinking we should reconsider how we notify the synchronizer. Since the synchronizer is a background process, I'm thinking instead of passing sync directly to all the other components of the daemon, maybe we pass a sync channel to all the components, so each component pushes the data they want to the synchronizer.

I'm saying this because even a `if tc.sync != nil` does not guarantee that before `tc.sync.NotifyBucketStartup()` is called, `tc.sync` would not have been nil.

Reply:
@perfectmak 

I fully agree with passing the `chan` around instead of an object.